include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -209,35 +242,73 @@ inline void StatsdClient::count(const std::string& key,
+                                const float frequency,
+                                const std::vector<std::string> &tags) const noexcept {
+    send(key, value, detail::METRIC_TYPE_GAUGE, frequency, tags);
+}
+
+inline void StatsdClient::gauge(const std::string &key,
                                 const unsigned int value,
                                 const float frequency,
-                                const std::vector<std::string>& tags) const noexcept {
-    return send(key, value, detail::METRIC_TYPE_GAUGE, frequency, tags);
+                                const std::vector<std::string> &tags) const noexcept {
+    send(key, value, detail::METRIC_TYPE_GAUGE, frequency, tags);
+}
+
+inline void StatsdClient::gauge(const std::string &key,
+                                const float value,
+                                const float frequency,
+                                const std::vector<std::string> &tags) const noexcept {
+    send(key, value, detail::METRIC_TYPE_GAUGE, frequency, tags);
 }
 
 inline void StatsdClient::timing(const std::string& key,
                                  const unsigned int ms,
                                  float frequency,
                                  const std::vector<std::string>& tags) const noexcept {
-    return send(key, ms, detail::METRIC_TYPE_TIMING, frequency, tags);
+    send(key, ms, detail::METRIC_TYPE_TIMING, frequency, tags);
 }
 
 inline void StatsdClient::set(const std::string& key,
                               const unsigned int sum,
                               float frequency,
                               const std::vector<std::string>& tags) const noexcept {
-    return send(key, sum, detail::METRIC_TYPE_SET, frequency, tags);
+    send(key, sum, detail::METRIC_TYPE_SET, frequency, tags);
+}
+
+inline void StatsdClient::send(const std::string &key,
+                               const unsigned int value,
+                               const char *type,
+                               float frequency,
+                               const std::vector<std::string> &tags) const noexcept {
+    send(key, std::to_string(value), type, frequency, tags);
 }
 
-inline void StatsdClient::send(const std::string& key,
+inline void StatsdClient::send(const std::string &key,
                                const int value,
-                               const char* type,
+                               const char *type,
+                               float frequency,
+                               const std::vector<std::string> &tags) const noexcept {
+    send(key, std::to_string(value), type, frequency, tags);
+}
+
+inline void StatsdClient::send(const std::string &key,
+                               const float value,
+                               const char *type,
+                               float frequency,
+                               const std::vector<std::string> &tags) const noexcept {
+    send(key, std::to_string(value), type, frequency, tags);
```

Comment:
@messiaen rather than add `to_string` everywhere could we just `template` the send function so it could take either float or int?

Reply:
Using template now.